README.md

```diff
@@ -327,6 +330,89 @@ The rough heuristic here is if we have custom arguments:
    environment variables. (Builder uses `_BUILDER_ARGS_CUSTOM_FLAGS`).
 2. If a non-`builder` command, then append without ` -- ` token.
 
+###### Expanding the Archetype Path
+
+Builder tasks often refer to configuration files in the archetype itself like:
+
+```js
+"postinstall": "webpack --bail --config node_modules/<archetype>/config/webpack/webpack.config.js",
+```
+
+In npm v2 this wasn't a problem because dependencies were usually nested. In
+npm v3, this all changes with aggressive
+[flattening](https://docs.npmjs.com/cli/dedupe) of dependencies. With flattened
+dependencies, the chance that the archetype and its dependencies no longer have
+a predictable contained structure increases.
+
+Thus, commands like the above succeed if the installation ends up like:
+
+```
+node_modules/
+  <a module>/
+    node_modules/
+      <archetype>/
+        node_modules/
+          webpack/
+```
+
+If npm flattens the tree like:
+
+```
+node_modules/
+  <a module>/
+  <archetype>/
+  webpack/
+```
+
+Then `builder` can still find `webpack` due to its `PATH` and `NODE_PATH`
+mutations. But an issue arises with something like a `postinstall` step after
+this flattening in that the current working directory of the process will be
+`PATH/TO/node_modules/<a module>/`, which in this flattened scenario would
+**not** find the file:
+
+```
+node_modules/<archetype>/config/webpack/webpack.config.js
+```
+
+because relative to `node_modules/<a module>/` it is now at:
+
+```
+../<archetype>/config/webpack/webpack.config.js
+```
+
+To address this problem `builder` has an `--expand-archetype` flag that will
+replace an occurrence of the specific `node_modules/<archetype>` in one of the
+archetype commands with the _full path_ to the archetype, to guarantee
+referenced files are correctly available.
+
+The basic heuristic of things to replace is:
+
+* `^node_modules/<archetype>`: Token is very first string.
+* `[\s\t]node_modules/<archetype>`: Whitespace before token.
+* `['"]node_modules/<archetype>`: Quotes before token.
```

Comment:
Great to see a solution to `victory` postinstall woes! I'm wondering if it might be better to require a specific token in the archetype script command, though, rather than doing magic inferences with `node_modules/<archetype>`?


e.g. Victory's `"build-dist-min"` command could change from

```
"webpack --bail --config node_modules/builder-victory-component/config/webpack/webpack.config.js --colors"
```
to something like
```
"webpack --bail --config `<%= ARCHETYPE_PATH %>/config/webpack/webpack.config.js --colors"
```

(not sure it makes sense to use the `bulider-init` template variable syntax, but I think you get the point).

In this case, when Builder encounters that token, it could auto-replace that with the path to the archetype, removing the need for a special flag (`--expand-archetype`).